wire-ios-data-model/Source/ManagedObjectContext/Migration/2.112-2.113/OneOnOneConversationMigrationPolicy.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+// Up until model version 2.112, a user was related to their one on one
+// conversation via the `connection` relationship, ie `user.connection.conversation`
+// and inversely `conversation.connection.to`.
+
+class OneOnOneConversationMigrationPolicy: NSEntityMigrationPolicy {
```

Comment:
Wow good job here! thanks!

Reply:
It took a while to understand the nuances of how to do heavyweight migrations 😅